dali/kernels/audio/mel_scale/mel_filter_bank_gpu.cu

```diff
@@ -229,31 +510,29 @@ class MelFilterBankGpu<T>::Impl : public MelFilterImplBase<T> {
           block_descs_[block_id].out_frame = out;
           ++block_id;
         }
-        in += nwindows_[ti] * fft_dim_;
+        in += nwindows_[ti] * nfft_;
         out += nwindows_[ti] * args_.nfilter;
       }
     }
   }
 
-  void FillBlockDescsInnerFft(const T* const* in_list, T **out_list) {
-    int block_id = 0;
-    int sample = 0;
-    while (block_id < static_cast<int>(block_descs_.size())) {
-      auto sample_size = block_descs_[block_id].out_frame_size;
-      auto nblocks = div_ceil(sample_size, kBlockDim1);
-      for (int i = 0; i < nblocks; ++i, ++block_id) {
-        block_descs_[block_id].in_frame = in_list[sample];
-        block_descs_[block_id].out_frame = out_list[sample];
-      }
-      ++sample;
+  void FillBlockDescsInnerFft(T *const *out_list, const T *const *in_list) {
+    for (size_t block_id = 0; block_id < block_descs_.size(); block_id++) {
+      int sample = block2sample_[block_id];
+      block_descs_[block_id].SetBasePointers(out_list[sample], in_list[sample]);
     }
   }
 
   std::vector<int> interval_ends_;
+  std::vector<int> fft2mel_;
+  std::vector<T> weights_down_norm_, weights_up_norm_;
   std::vector<int64_t> nframes_;
   std::vector<int64_t> nwindows_;
   std::vector<BlockDesc<T>> block_descs_;
-  int64_t fft_dim_ = 0;
+  std::vector<int> block2sample_;
+  int64_t nfft_ = 0;
+  int shm_height_ = 32;
+  int shm_size_ = 32 << 10;
```

Comment:
perhaps we could put those tuneable constants (shm_height, shm_size, windows_per_block) together with a note that those were selected to maximize performance (just in case we need to optimize in the future again)

Reply:
Those are just defaults - I could leave them at invalid values here, since they'll be overwritten anyway.